### PR TITLE
add test case to push/pull edge image in quay.io

### PR DESCRIPTION
**Pull request by yih-redhat**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

Add test case to push/pull rhel for edge container image in quay.io.
The destination quay repository is docker://quay.io/osbuild/testing-rhel-edge-push
Find the repository details and credentials in https://issues.redhat.com/browse/COMPOSER-816

### CODE REVIEW

File: test/cases/ostree-ng.sh [hunk 1]
```diff
@@ -13,6 +13,9 @@ function greenprint {
     echo -e "\033[1;32m${1}\033[0m"
 }
 
+# Install skopeo 
+sudo dnf install -y skopeo
```
Comment (atodorov) [inline test/cases/ostree-ng.sh:17]: I think we should add this to osbuild-composer.spec, requirements for the -tests package.

**Review by atodorov — CHANGES_REQUESTED** [review]

File: test/cases/ostree-ng.sh (hunk 1 above)
Comment (yih-redhat) [reply test/cases/ostree-ng.sh]: Thank you. Added skopeo in osbuild-composer.spec - requirements.

**Comment by yih-redhat** [comment]
> In Jenkinsfile, for the stages running ostree-ng add the following:
> 
> ```
> environment {
>     QUAY_USERNAME = credentials('quay-username')
>     QUAY_PASSWORD = credentials('quay-password')
> }
> ```

Hi Alex, I have updated schutzbot/jenkins file, now it looks like:
    if (test_type == 'ostree-ng') {
        environment {
            QUAY_USERNAME = credentials('quay-username')
            QUAY_PASSWORD = credentials('quay-password')
        }
        sh (
            label: "New OSTree tests",
            script: "/usr/libexec/tests/osbuild-composer/ostree-ng.sh"
        )
    }

File: schutzbot/Jenkinsfile [hunk 2]
```diff
@@ -807,6 +807,10 @@ void run_tests(test_type) {
     }
 
     if (test_type == 'ostree-ng') {
+        environment {
+            QUAY_USERNAME = credentials('quay-username')
+            QUAY_PASSWORD = credentials('quay-password')
+        }
```
Comment (atodorov) [inline schutzbot/Jenkinsfile:813]: That's not going to work, it's not a valid syntax. See https://www.jenkins.io/doc/book/pipeline/syntax/#environment and https://github.com/osbuild/osbuild-composer/pull/1400#pullrequestreview-656413994 and the rest of Jenkinsfile.

**Review by atodorov — CHANGES_REQUESTED** [review]

**Comment by yih-redhat** [comment]
> @yih-redhat what is your username on GitLab.com ?

I used github account to login gitlab, my username in gitlab is yih-redhat

**Comment by atodorov** [comment]
blocked by #1408 b/c we need to see the test reporting PASS.

**Comment by atodorov** [comment]
@yih-redhat you can now rebase to the latest main branch and this should be ready to go.
